File: LeetCode/remote/plus_one.rb
#   https://leetcode.com/problems/plus-one/
#   Given a non-negative number represented as an array of digits, plus one to the number.
#   https://leetcode.com/submissions/detail/54458870/

# @param {Integer[]} digits
# @return {Integer[]}
def plus_one(digits)
    remainder = 1
    (digits.size - 1).downto(0).each { |i|
        digits[i] += remainder
        remainder = digits[i] / 10
        digits[i] %= 10
    }
    
    digits.unshift(remainder) if remainder != 0
    return digits
end

File: LeetCode/remote/plus_one.cs
//  https://leetcode.com/problems/plus-one/
//  https://leetcode.com/submissions/detail/66847531/
//
//  Submission Details
//  108 / 108 test cases passed.
//      Status: Accepted
//      Runtime: 460 ms
//          
//          Submitted: 0 minutes ago
public class Solution {
    public int[] PlusOne(int[] digits) {
        var carry = 1;
        var i = digits.Length - 1;
        while (carry != 0 && i >= 0) {
            carry = Math.DivRem(digits[i] + carry, 10, out digits[i]);
            i--;
        }
        
        return carry == 0 ? digits : new [] { carry }.Concat(digits).ToArray();
    }
}
